pkg/ipamd/ipamd.go

```diff
@@ -942,13 +942,12 @@ func (c *IPAMContext) tryAssignIPs() (increasedPool bool, err error) {
 				return false, errors.Wrap(err, fmt.Sprintf("failed to allocate one IP addresses on ENI %s, err ", eni.ID))
 			}
 		}
-		// This call to EC2 is needed to verify which IPs got attached to this ENI.
-		ec2Addrs, err := c.awsClient.GetIPv4sFromEC2(eni.ID)
+		eniMetadata, err := c.awsClient.WaitForENIAndIPsAttached(eni.ID, resourcesToAllocate)
```

Comment:
resourcesToAllocate seems to be new IPs to allocate.

1. If there are already 8IPs, and we allocate 3 more. this wait will immediately succeed. (and the 3 more additional ip won't be added to pool).
2. what happens if the imds wait timeout, do we have any code that reconciles ip pool on eni at later time?

Reply:
Reg 1. `ec2Addrs := eniMetadata.IPv4Addresses` should have all the IPs include the new 3 IPs right? Is my understanding correct?

2. yes the reconciler fetches from IMDS and adds those new IPs to datastore.

Reply:
@M00nF1sh actually it should be (resourcesToAllocate + currentNumberOfAllocatedIPs) for the wait.